incubator/binding-mqtt/src/test/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/PublishIT.java

```diff
@@ -0,0 +1,322 @@
+    }
+
+    @Test
+    @Configuration("server.when.topic.publish.only.yaml")
+    @Specification({
+        "${net}/publish.one.message/client",
+        "${app}/publish.one.message/server"})
+    @Configure(name = WILDCARD_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = SHARED_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = MAXIMUM_QOS_NAME, value = "2")
+    @Configure(name = SESSION_EXPIRY_INTERVAL_NAME, value = "0")
+    public void shouldPublishOneMessageViaRoute() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Configuration("server.yaml")
+    @Specification({
+        "${net}/publish.retained/client",
+        "${app}/publish.retained/server"})
+    @Configure(name = WILDCARD_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = SHARED_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = MAXIMUM_QOS_NAME, value = "2")
+    @Configure(name = SESSION_EXPIRY_INTERVAL_NAME, value = "0")
+    public void shouldPublishRetainedMessage() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Configuration("server.yaml")
+    @Specification({
+        "${net}/publish.message.with.topic.alias/client",
+        "${app}/publish.message.with.topic.alias/server"})
+    @Configure(name = WILDCARD_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = SHARED_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = MAXIMUM_QOS_NAME, value = "2")
+    @Configure(name = TOPIC_ALIAS_MAXIMUM_NAME, value = "2")
+    @Configure(name = SESSION_EXPIRY_INTERVAL_NAME, value = "0")
+    public void shouldPublishMessageWithTopicAlias() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Configuration("server.yaml")
+    @Specification({
+        "${net}/publish.multiple.messages/client",
+        "${app}/publish.multiple.messages/server"})
+    @Configure(name = WILDCARD_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = SHARED_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = MAXIMUM_QOS_NAME, value = "2")
+    @Configure(name = SESSION_EXPIRY_INTERVAL_NAME, value = "0")
+    public void shouldPublishMultipleMessages() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Configuration("server.yaml")
+    @Specification({
+        "${net}/publish.multiple.messages.with.delay/client",
+        "${app}/publish.multiple.messages/server"})
+    @Configure(name = WILDCARD_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = SHARED_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = MAXIMUM_QOS_NAME, value = "2")
+    @Configure(name = SESSION_EXPIRY_INTERVAL_NAME, value = "0")
+    @Configure(name = PUBLISH_TIMEOUT_NAME, value = "5")
+    public void shouldPublishMultipleMessagesWithDelay() throws Exception
+    {
+        k3po.start();
+        k3po.awaitBarrier("PUBLISHED_MESSAGE_TWO");
+        Thread.sleep(1500);
```

Comment:
Can we reduce the delay?

Reply:
Actually is this test usefully different from immediate publish of the second message?

Perhaps a better test would be to let the publish app stream timeout and verify that it cleans up then another message to the same topic creates a new publish app stream?

Reply:
I suppose the benefit of this test is to verify that the publish stream didn't close immediately, and that it can be reused over time.